server/src/controllers/AuthenticationController.ts

```diff
@@ -0,0 +1,42 @@
+import "reflect-metadata";
+import * as restify from "restify";
+import { Controller, interfaces, Post } from "inversify-restify-utils";
+import { injectable } from "inversify";
+import { AuthenticationService } from "../services/AuthenticationService";
+import * as Joi from "joi";
+
+@Controller("/")
+@injectable()
+export class AuthenticationController implements interfaces.Controller {
+    private readonly _authenticationService: AuthenticationService;
+
+    constructor(_authService: AuthenticationService) {
+        this._authenticationService = _authService;
+    }
```

Comment:
You can then inject the class this way. Do this similar thing in the other classes as well

```suggestion
    constructor(
        @inject('Service') @named('AuthenticationService') private readonly _authService: AuthenticationService
    ) {}
```